### PR TITLE
resolver use package-lock.json or yarn lock

### PR DESCRIPTION
**Pull request by yann300**
- url resolver now resolves dependencies using the version number provided by yarn.lock or package-lock.json.
 - remove uneeded cache.
 - fix resolving the package.json dep.

### CODE REVIEW

**Review by bunsenstraat — APPROVED** [review]

File: libs/remix-url-resolver/src/resolve.ts [hunk 1]
```diff
@@ -127,11 +135,51 @@ export class RemixURLResolver {
...
+              if (version) url = url.replace(pkg, `${pkg}@${version}`)
+            }
+          }
+        } catch (e) {
+          console.log(e)
+        }
+      }
       const req = 'https://unpkg.com/' + url
       const response: AxiosResponse = await axios.get(req, { transformResponse: [] })
+      console.log('handleNpmImport', req)
```
Comment (ioedeveloper) [inline libs/remix-url-resolver/src/resolve.ts:182]: Do you need this console.log?

File: libs/remix-core-plugin/src/lib/compiler-content-imports.ts [hunk 2]
```diff
@@ -16,12 +16,39 @@ export type ResolvedImport = {
...
+      try {
+        let yarnLock
+        if (await this.call('fileManager', 'exists', './yarn.lock')) {
+          yarnLock = await this.call('fileManager', 'readFile', './yarn.lock')
+        }
+
+        let packageLock
+        if (await this.call('fileManager', 'exists', './package-lock.json')) {
+          packageLock = await this.call('fileManager', 'readFile', './package-lock.json')
+          packageLock = JSON.parse(packageLock)
```
Comment (ioedeveloper) [inline libs/remix-core-plugin/src/lib/compiler-content-imports.ts:32]: Needs to be tested with a large `package-lock.json` file

Comment (bunsenstraat) [reply libs/remix-core-plugin/src/lib/compiler-content-imports.ts:32]: Very true. We need an example.

**Comment by bunsenstraat** [comment]
I tested it with a package file but the behavior is inconsistent. Needs some debugging.

**Pull request merged**